morpheus/stages/inference/triton_inference_stage.py

```diff
@@ -702,28 +707,51 @@ def __init__(self,
                  force_convert_inputs: bool = False,
                  use_shared_memory: bool = False,
                  needs_logits: bool = None,
-                 inout_mapping: dict[str, str] = None):
+                 inout_mapping: dict[str, str] = None,
+                 input_mapping: dict[str, str] = None,
+                 output_mapping: dict[str, str] = None):
         super().__init__(c)
 
         self._config = c
 
         if needs_logits is None:
             needs_logits = c.mode == PipelineModes.NLP
 
-        # Combine the pipeline mode defaults with any user supplied ones
-        inout_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {})
+        input_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {}).get("inputs", {})
+        output_mapping_ = self._INFERENCE_WORKER_DEFAULT_INOUT_MAPPING.get(c.mode, {}).get("outputs", {})
+
+        inout_mapping_ = {}
+        inout_mapping_.update(input_mapping_)
+        inout_mapping_.update(output_mapping_)
+
         if inout_mapping is not None:
             inout_mapping_.update(inout_mapping)
+            input_mapping_.update(inout_mapping)
+            output_mapping_.update(inout_mapping)
+
+        if input_mapping is not None:
+            input_mapping_.update(input_mapping)
+
+        if output_mapping is not None:
+            output_mapping_.update(output_mapping)
 
         self._kwargs = {
-            "model_name": model_name,
             "server_url": server_url,
+            "model_name": model_name,
             "force_convert_inputs": force_convert_inputs,
             "use_shared_memory": use_shared_memory,
             "inout_mapping": inout_mapping_,
             "needs_logits": needs_logits
         }
 
+        self._server_url = server_url
+        self._model_name = model_name
+        self._force_conver_inputs = force_convert_inputs
+        self._use_shared_memory = use_shared_memory
+        self._input_mapping = input_mapping_
+        self._output_mapping = output_mapping_
+        self._needs_logits = needs_logits
```

Comment:
This is better than `self._kwargs`. But we dont need both. Remove `self._kwargs`